includes/wikia/api/DesignSystemApiController.class.php

```diff
@@ -1,21 +1,185 @@
+					[
+						'type' => 'link-branded',
+						'brand' => 'wikis',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-vertical-wikis'
+						],
+						'href' => '#'
+					],
+					[
+						'type' => 'link-branded',
+						'brand' => 'communitycentral',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-vertical-communitycentral'
+						],
+						'href' => '#'
+					]
+				]
+			]
+		],
+		'search' => [
+			[
+				'type' => 'search-endpoint',  // new_type
+				'endpoint-href' => 'some.search.link',
+				'suggestions-href' => 'suggestions.link', // make object
+				'placeholder-inactive' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-search-placeholder-inactive'
+				],
+				'placeholder-active' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-search-placeholder-active'
+				],
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-search-title'
+				]
+			]
+		],
+		'user-info' => [
+			'type' => 'user-not-authenticated', // new_type
+			'avatar' => [ ],
+			'links' => [
+				[
+					'type' => 'link-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-signin-title'
+					],
+					'href' => '#'
+				],
+				[
+					'type' => 'link-text',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-register-title'
+					],
+					'description' => [  // define_new_type?
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-register-description'
+					],
+					'href' => '#'
+				]
+			]
+		],
+		'start-wikia' => [
+			'type' => 'link-text',
+			'title' => [
+				'type' => 'translatable-text',
+				'key' => 'wikia-create-wiki-link-start-wikia'
+			],
+			'href' => '#'
+		]
+	];
+
 	public function getFooter() {
+		$this->checkRequestCompleteness();
+
+		$footerModel = new DesignSystemGlobalFooterModel( $this->wikiId, $this->lang );
+
+		$this->setResponseData( $footerModel->getData() );
+		$this->response->setCacheValidity( WikiaResponse::CACHE_VERY_SHORT );
+	}
+
+	public function getNavigation() {
+		global $wgUser;
+		$this->checkRequestCompleteness();
+
+		// TODO: change to not mocked data
+		$this->setResponseData( $this->data );
+
+		if ( $wgUser) {
+			$this->response->setCachePolicy( WikiaResponse::CACHE_PRIVATE );
+			$this->response->setCacheValidity( WikiaResponse::CACHE_DISABLED );
+		} else {
+			$this->response->setCacheValidity( WikiaResponse::CACHE_VERY_SHORT );
+		}
+	}
+
+	private function checkRequestCompleteness() {
 		$wikiId = $this->request->getInt( 'wikiId' );
 		$lang = $this->request->getVal( 'lang' );
 
 		if ( WikiFactory::IDtoDB( $wikiId ) === false ) {
-			throw new NotFoundApiException( "Unable to find wiki with ID {$wikiId}" );
+			throw new NotFoundApiException( 'Unable to find wiki with ID {$wikiId}' );
```

Comment:
I think you will get `{$wikiId}` instead of actual wiki id when using single apostrophes.